### ISSUE TITLE
Callout blocks

### ISSUE THREAD

**Issue by jasloe** [aarome/aarome#290]
There are a number of callout blocks scattered throughout the site, e.g.:

```
By appointment, Monday-Friday: 14:30 – 16:30
```

and

```
Tuesday, Thursday:

10:00 – 13:00/13:30 – 17:00

or by appointment
```

These were designed to live at the bottom of the page which no one will ever read. I made the executive decision to move these to the space between the bottom of the image and the top of the body copy.

@cchristopherhowardd –– when you have a minute can you introduce some text that reflects the current "situation"?

**Comment by cchristopherhowardd** [comment]
Do you have a list of pages on which these callout blocks appear? I recall the Research section having some of them. I want to be thorough in my review.

**Comment by jasloe** [comment]
@cchristopherhowardd ––

I've added block editing privileges to the 'site administrator' account.

Callouts are stored in what Drupal terms "blocks." 

From toolbar go to ```Structure/Block Layout/Custom Block Library```. 

The resulting list contains the current set of blocks in use throughout the site. These are primarily related to library reference desk and resource hours.

Note that just like nodes, they will require a suitable Italian-language translation.

I am working on image handling issues today and tomorrow. I'll work on finishing up the mechanism for attaching blocks to nodes in that timeframe.

**Comment by cchristopherhowardd** [comment]
I have updated all of the appropriate blocks. It seems like the style depends on limiting the length of the text, so that the width of the block is constrained. Is this correct?

**Comment by jasloe** [comment]
Great. Block length can be any length. They'll appear in a call out about the first paragraph on respective page. It'll look good.

**Comment by cchristopherhowardd** [comment]
OK. I was wondering which would look better: fragmented phrasing or complete sentences.

**Comment by cchristopherhowardd** [comment]
The callout blocks aren't working the way they should, visually (how they look), spatially (where they're placed on the page), and functionally (how they work). I'm reviewing Opens designs from 2018 and locating the pages where they're supposed to appear.

**Comment by jasloe** [comment]
I was experimenting with placement and editorial workflow. No one is going to read/see a block at the bottom of a page. Tree falls, etc. 

I'll revert to the original design.

**Comment by cchristopherhowardd** [comment]
For instance, here is Open's design for callout blocks on library webpages. The idea was to place the hours for each branch in the same block, with a general text about closings underneath it. Your comment above was that the block lives at the bottom of the page where nobody sees it. Maybe that's true, but if the block appeared on all Research pages, then it becomes ambient, familiar, and easily accessible to anyone using the Research section. A person doesn't have to hunt down the page with information about the Photographic Archive or the AAR Library, which we've placed on singular, unconnected webpages.

![Laptop 1080 – Overview 7 - Resources](https://user-images.githubusercontent.com/38765846/82484530-0a86a180-9aa8-11ea-97f8-bdce49fb34f0.png)

**Comment by cchristopherhowardd** [comment]
When you write "No one is going to read/see a block at the bottom of a page," I have to disagree. Last month I had to deal with a colleague who was steadfastly determined that "nobody scrolls" and that "our people don't scroll" when that is 100 percent not true. She wanted all of the relevant information for a webpage to appear on the screen all at once, so that our people did not have to scroll. Myths about placing stuff above the fold is years out of date, if not decades. Anyone who says "nobody scrolls" has probably never heard of Twitter or Instagram or Facebook. Or smart phones. Or the internet.

**Comment by jasloe** [comment]
Original pattern restored. Working on digging up the callout template to add Visiting Hours and the holiday caption back in.

https://staging.aarome.org/research/resources/online-catalogues

**Comment by cchristopherhowardd** [comment]
That link reminds me that I really need to replace the image on that page. Ugly and confusing.

**Comment by cchristopherhowardd** [comment]
The Society of Fellows wants a callout block on their pages. Can the following phrase be placed on every page that begins with aarome.org/society-of-fellows/?

To receive news and updates for the Society of Fellows, please send a message to SOF_communications@aarome.org with the subject line “Email list.” 

Obviously https://staging.aarome.org/society-of-fellows/news-events won’t have a callout block, but individual news and events pages (like https://staging.aarome.org/society-of-fellows/news-events/private-tour-brant-foundation and https://staging.aarome.org/society-of-fellows/news-events/sof-news-february-2018 ) would.

The language in the block will likely change. Nobody provided an exact phrase, so I wrote what I wrote.

**Comment by jasloe** [comment]
Meaning callout blocks can be placed on a per-node basis or, instead, on all SOF news and events? (I think you're describing the latter; just want to be sure.)

**Comment by cchristopherhowardd** [comment]
Well, we don't have many SOF pages, so the callout box can be placed on the Past Presidents or Bylaws page on a per node basis, right? And the process for adding callout blocks for all SOF News & Events page would be automatic?

**Comment by cchristopherhowardd** [comment]
In the News section, the Features pages don’t need the callout box. Occasionally a Feature will be an official AAR announcement (such as https://staging.aarome.org/news/features/avinoam-shalem-appointed-24th-director-american-academy-rome), but contact information will appear in the running text.

The callout block on the Press Releases page looks good: https://staging.aarome.org/news/press-releases

**Comment by jasloe** [comment]
Yes, for SOF/Events it's managed in configuration and would appear on all pages. (You can change the text through the custom block editor, just like library hours, et al.). I'll get started on that. 

For the other use case, there are a couple of ways to go about it depending on how much flexibility is required:

1. Hard code a reference to the unique block id in a template file:
```
		{% if 'past-presidents' in url %}
			{{ drupal_entity('block_content', some-block-id-number) }}
		{% endif %}
```
This would be managed in the theme code. 

2. Or, we can add a module called block field that let's the user pick which block they want to appear on a node. 

This is Academy-managed.

1. is easiest (good), doesn't require a module (better), and is more or less foolproof (best).
2. is another module (not so good), adds another field to the node editor (eh), but is the most flexible (best).

I vote for 1. If the need arises, we can revisit the second option.

**Comment by jasloe** [comment]
Ok, noted.

Sorry, can you add the Open file for Press Inquiries to this ticket? I don't have it handy.

For press, I'll change the "Press inquiries" header to serif. Isn't your name supposed to appear there, too? No problem to add the addresses.



> In the News section, the Features pages don’t need the callout box. Occasionally a Feature will be an official AAR announcement (such as https://staging.aarome.org/news/features/avinoam-shalem-appointed-24th-director-american-academy-rome), but contact information will appear in the running text.
> 
> The callout block on the Press Releases page looks good: https://staging.aarome.org/news/press-releases. Open's design called for listing the New York and Rome street addresses, but that's unnecessary.

**Comment by jasloe** [comment]
Do we have a translation for Press Inquiries and Visiting Hours? Any other callouts needed or in the works?

**Comment by cchristopherhowardd** [comment]
I vote for 1, too. I'll translate Press Inquiries and Visiting Hours tomorrow. Still working through where all callouts go and if we need any more. SOF was the only new callout request.

**Comment by cchristopherhowardd** [comment]
The security notice is a callout block, is it not? See the bottom of https://staging.aarome.org/events/calendar/canceled-matthew-brennan-james-casebere or most other Events pages (especially those at the Academy in Rome). I make a selection in the back end of Event pages whether or not to include the security notice.

<img width="942" alt="Screen Shot 2020-05-21 at 10 39 54 AM" src="https://user-images.githubusercontent.com/38765846/82570418-8ee02f80-9b4f-11ea-8e5a-26f752d4e400.png">

I imagine that the security notice should be in a green box, like other callouts, right? Open didn't include a callout for the security notice in their original design because they assume this notice would appear in the running text (which is does on the current site). We made a decision last year to bake the notice into the page.

**Comment by jasloe** [comment]
Yes, the library block uses a slightly different template due to the way it is administered. 

https://staging.aarome.org/events/calendar/canceled-matthew-brennan-james-casebere

I think the saner way to do this is show the block based on locality, i.e. if the place reference == American Academy in Rome then show block. 

But this will work for now. 

Fixed h2 styling for news/press callout. 

https://staging.aarome.org/news/features/memoriam-nancy-o-boyle

Add your name and Academy addresses or no?

(Note I temporarily removed image styling for news/features to work on #298. Included a note on that in that ticket.)

**Comment by cchristopherhowardd** [comment]
For the News callout block, add the Rome person:

Maddalena Bonicelli
Rome Press Officer
+39 335 6857707
m.bonicelli.ext@aarome.org

**Comment by jasloe** [comment]
Added.

**Comment by cchristopherhowardd** [comment]
Any progress yet for the SOF section callout block? I suggested the running text of: To receive news and updates for the Society of Fellows, please send a message to SOF_communications@aarome.org with the subject line “Email list.”

If the block needs a header, it can read: Join the SOF Email List

**Comment by cchristopherhowardd** [comment]
UPDATE! The callout box for SOF pages does not need to appear on News & Events pages ( https://staging.aarome.org/society-of-fellows/news-events ). They only need to appear on the other SOF pages, of which there are five.

**Comment by cchristopherhowardd** [comment]
Callout blocks for the News section should only have the names and contact information. AAR's two addresses should be removed.

<img width="1386" alt="Screen Shot 2020-06-09 at 1 56 28 PM" src="https://user-images.githubusercontent.com/38765846/84183037-4c788700-aa59-11ea-88a7-f49898c98489.png">

**Comment by jasloe** [comment]
Removed.
